src/Lib.P2P/SecureCommunication/Psk1Stream.cs

```diff
@@ -0,0 +1,203 @@
+                    _readCipher.Init(false, new ParametersWithIV(new KeyParameter(_key.Value), nonce));
+                }
+
+                return _readCipher;
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool CanRead => _stream.CanRead;
+
+        /// <inheritdoc />
+        public override bool CanSeek => false;
+
+        /// <inheritdoc />
+        public override bool CanWrite => _stream.CanWrite;
+
+        /// <inheritdoc />
+        public override bool CanTimeout => _stream.CanTimeout;
+
+        /// <inheritdoc />
+        public override long Length => throw new NotSupportedException();
+
+        /// <inheritdoc />
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override void SetLength(long value) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            var cipher = ReadCipher;
+            var n = _stream.Read(buffer, offset, count);
+            cipher.ProcessBytes(buffer, offset, n, buffer, offset);
+            return n;
+        }
+
+        /// <inheritdoc />
+        public override async Task<int> ReadAsync(byte[] buffer,
+            int offset,
+            int count,
+            CancellationToken cancellationToken)
+        {
+            var cipher = ReadCipher;
+            var n = await _stream.ReadAsync(buffer, offset, count, cancellationToken).ConfigureAwait(false);
+            cipher.ProcessBytes(buffer, offset, n, buffer, offset);
+            return n;
+        }
+
+        /// <inheritdoc />
+        public override void Flush() { _stream.Flush(); }
+
+        /// <inheritdoc />
+        public override Task FlushAsync(CancellationToken cancel) { return _stream.FlushAsync(cancel); }
+
+        /// <inheritdoc />
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            var x = new byte[count];
+            WriteCipher.ProcessBytes(buffer, offset, count, x, 0);
+            _stream.Write(x, 0, count);
+        }
+
+        /// <inheritdoc />
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            var x = new byte[count];
+            WriteCipher.ProcessBytes(buffer, offset, count, x, 0);
+            return _stream.WriteAsync(x, 0, count, cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public override void WriteByte(byte value) { _stream.WriteByte(WriteCipher.ReturnByte(value)); }
+
+        /// <inheritdoc />
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing) _stream.Dispose();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)